csrc/ops/utils.cpp

```diff
@@ -174,12 +174,128 @@ IterType promoteIterType(IterType type1, IterType type2) {
   }
 }
 
+//! For MatmulOp, the input iterdomains at a given index do not necessarily map to the output iterdomain at that index
+//! This function aligns the input iterdomain to the output and returns a vector where each element is the
+//! input iterdomain corresponding to the output iterdomain at that index.
+//! If the element is nullptr, there is no mapping between input-output at that
+//! index. Based on the input dimensions following cases are possible:
+//! 1. A/B is 1D: [M, K] x [K] -> [M]
+//! Mapping A: {id_M}, Mapping B: {nullptr}
+//! 2. A and B are 2D: [M, K] x [K, N] -> [M, N]
+//! Mapping A: {id_M, nullptr}, Mapping B: {nullptr, id_N}
+//! 3. A/B are atleast 1D and one of them is > 2D: [B, M, K] x [K, N] -> [B, M,
+//! N] Mapping A: {id_B, id_M, nullptr}, Mapping B: {nullptr, nullptr, id_N}
+std::vector<IterDomain*> mapMatmulOpIterDomains(
+    const std::vector<IterDomain*>& input_domain,
+    bool is_lhs,
```

Comment:
I'd use MmaOperand::A or ::B instead: https://github.com/NVIDIA/Fuser/blob/3598b4df9a2de7f9e202f93fb6f836c9168a402b/csrc/mma_type.h#L208

1. Enum is logically closer to what you are really trying to represent here. 
2. For better type safety. 
3. For better readability on the caller side. You'd need to write `/*is_lhs=*/true` and `/*is_rhs=*/false` in the caller to achieve similar readability; otherwise, I've no idea what `true/false` refers to at the first glance. 
4. Flexibility **if** you want this to be used for linear which takes three operands or for the destination matrix. But be aware of YAGNI. 

I was trying to search for some articles talking about enum vs boolean. https://betterprogramming.pub/dont-use-boolean-arguments-use-enums-c7cd7ab1876a seems to be a good reference.